### ISSUE TITLE
使用train_sft_with_dev_set.sh进行 lora的微调 如何平衡 'loss' 和 'eval_loss' 

### ISSUE THREAD

**Issue by mawenju203** [hiyouga/ChatGLM-Efficient-Tuning#369]
CUDA_VISIBLE_DEVICES=0 python ../src/train_bash.py \
    --model_name_or_path /data/mawj/ChatGLM2-6B/chatglm2-6b \
    --use_v2 true \
    --stage sft \
    --do_train \
    --dataset BIGV.train.history \
    --dataset_dir ../data \
    --finetuning_type lora \
    --output_dir path_to_sft_BIGV_histroy_checkpoint \
    --overwrite_cache \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 16 \
    --gradient_accumulation_steps 2 \
    --lr_scheduler_type cosine \
    --evaluation_strategy steps \
    --save_strategy steps \
    --logging_steps 10 \
    --eval_steps 500 \
    --save_steps 500 \
    --learning_rate 1e-4 \
    --num_train_epochs 200.0 \
    --dev_ratio 0.01 \
    --load_best_model_at_end \
    --plot_loss \
    --fp16

这个是我的 train_sft_with_dev_set.sh  里面的信息；

现在出现 loss 在下降 ； eval_loss 却降不下去；不知道什么原因

**Comment by mawenju203** [comment]
![image](https://github.com/hiyouga/ChatGLM-Efficient-Tuning/assets/30227934/d675b2ac-1f0c-4a5c-a1a9-a352aa970244)
![image](https://github.com/hiyouga/ChatGLM-Efficient-Tuning/assets/30227934/06029553-58a3-4d50-8abc-834338c3f674)

如果训练结束，我如何选择一个较为恰当的 checkpoint呢？

**Comment by Godsing** [comment]
我也遇到相同的情况，训练 loss 一直在降，但是 eval_loss 却不降。

但是我发现，eval_loss 虽然不降，但是计算出来的指标（例如 ROUGE-L 和 BLEU）却提升了。所以我现在的做法是， 类似于“二分查找”，手动去寻找合适的 checkpoint。
